extensions/sql-database-projects/src/models/project.ts

```diff
@@ -900,22 +900,48 @@ export class Project implements ISqlProject {
 		itemGroup.appendChild(newFileNode);
 	}
 
-	private removeFileFromProjFile(path: string): void {
+	private async removeFileFromProjFile(path: string): Promise<void> {
 		const fileNodes = this.projFileXmlDoc!.documentElement.getElementsByTagName(constants.Build);
 		const preDeployNodes = this.projFileXmlDoc!.documentElement.getElementsByTagName(constants.PreDeploy);
 		const postDeployNodes = this.projFileXmlDoc!.documentElement.getElementsByTagName(constants.PostDeploy);
 		const noneNodes = this.projFileXmlDoc!.documentElement.getElementsByTagName(constants.None);
 		const nodes = [fileNodes, preDeployNodes, postDeployNodes, noneNodes];
 
 		let deleted = false;
+
+		// remove the <Build Include="..."> entry if there is one
 		for (let i = 0; i < nodes.length; i++) {
 			deleted = this.removeNode(path, nodes[i]);
 
 			if (deleted) {
-				return;
+				// still might need to add a <Build Remove="..."> node if this is an sdk style project
+				if (this.isSdkStyleProject) {
+					break;
+				} else {
+					return;
+				}
 			}
 		}
 
+		// if it's an sdk style project, we'll need to add a <Build Remove="..."> entry to remove this file if it's
+		// still included by a glob
+		if (this.isSdkStyleProject) {
+			// write any changes from removing an include node and get the current files included in the project
+			if (deleted) {
+				await this.serializeToProjFile(this.projFileXmlDoc);
+			}
+			const currentFiles = await this.readFilesInProject();
+
+			// only add a node to exclude the file if it's still included by a glob
```

Comment:
I wonder if we even need to do this check or just add the <Build Remove=...> directly. If the user got to this point it means the file is already part of the build items right?

Reply:
I think the scenario where we wouldn't need to add a Build Exclude node is if something like outsideFile.sql is being removed and it's a file outside the project folder that was added to the project like `<Build Include="..\otherFolder\outsideFile.sql" />` and it isn't included by another include glob. So when that Build Include entry for outsideFile.sql is removed, it wouldn't be in the currentFiles anymore and an exclude entry wouldn't need to be added for it.

Reply:
Another idea is to add a property to each FileEntry that says how it was included (explicitly listed vs. globbed) to simplify the removal logic.  It'd basically reduce the logic here to "if explicit inclusion, delete tag; if glob, add Remove tag.  Though I suppose it'd need a "both" category in case there's overlap (dunno why the user would do that, but it's possible).

Thoughts on that?  I'm fine either way.

Reply:
Hmm maybe that can be an enhancement we can look into later if we're looking at improving performance. 

Another case I thought of when a Build Remove wouldn't need to be added is if the file was deleted.